Rules/CommonScripts/PostGameMapVotes.as

```diff
@@ -0,0 +1,248 @@
+	if (this.get("MapVotesMenu", @mvm))
+	mvm.isSetup = false;
+	current_Selected = 0;
+}
+
+void onTick( CRules@ this )
+{		
+	MapVotesMenu@ mvm;
+	if (!this.get("MapVotesMenu", @mvm)) return;
+	if (!this.isGameOver()) return;
+	if (!mvm.isSetup)
+	{	
+		fadeTimer = -(4*30); // ticks of endgame time before fading
+		mvm.VoteTimeLeft = VoteSecs;
+		mvm.Refresh();
+
+		return;
+	}
+
+	if (fadeTimer < FadeTicks)
+	{
+		fadeTimer++;
+		return;
+	}
+	else if (fadeTimer == FadeTicks)
+	{
+		CPlayer@ player;
+		for (int i = 0; i < getPlayersCount(); i++)
+		{
+			@player = getPlayer(i);
+			CBlob@ blob = player.getBlob();
+
+			if (blob !is null)
+			{
+				blob.server_Die();				
+				getHUD().SetDefaultCursor();
+			}
+		}
+	}
+
+	// Vote is now setup, faded to black and is counting down
+	if (getGameTime() % 30 == 0)
+	mvm.VoteTimeLeft--;
+
+	if (mvm.VotedCount1 > mvm.VotedCount2 && mvm.VotedCount1 > mvm.VotedCount3)
+	{	//map 1 got the most votes
+		mvm.MostVoted = 1;
+	}
+	else if (mvm.VotedCount3 > mvm.VotedCount1 && mvm.VotedCount3 > mvm.VotedCount2)
+	{	//map 3 got the most votes
+		mvm.MostVoted = 3;
+	}
+	else 
+	{	//random map got the most votes or inconclusive
+		mvm.MostVoted = 2;
+	}
+
+	CBitStream params;
+	if (getNet().isServer() && mvm.VoteTimeLeft == -4) //timeup + 4 secs, load voted map
+	{
+		params.write_u8(mvm.MostVoted);
+		this.SendCommand(this.getCommandID(vote_end_id), params);
+	}
+
+	//---------- CLIENT -----------\\
+	CPlayer@ me = getLocalPlayer();
+	if (!getNet().isClient()) return;
+
+	CControls@ controls = getControls();
+	if (controls is null) return;
+
+	if (mvm.VoteTimeLeft <= 0 || mvm.VoteTimeLeft >= VoteSecs || me.getTeamNum() == this.getSpectatorTeamNum()) return;
+
+	u8 SelectedNum;	//default to zero, so command is sent only once
+	mvm.Update(controls, SelectedNum);
+
+	if (SelectedNum != 0)
+	{
+		u16 id = me.getNetworkID();
+		params.write_u16(id);
+		params.write_u8(SelectedNum);
+		params.write_u8(current_Selected);
+		this.SendCommand(this.getCommandID(vote_selectmap_id), params);
+	}
+}
+
+void onCommand(CRules@ this, u8 cmd, CBitStream@ params)
+{
+	MapVotesMenu@ mvm;
+	if (!this.get("MapVotesMenu", @mvm)) return;
+	
+	if (cmd == this.getCommandID(vote_selectmap_id))
+	{
+		u16 id = params.read_u16();
+		u8 selected = params.read_u8(); 
+		u8 lastselected = params.read_u8();
+		CPlayer@ player = getPlayerByNetworkId(id);	
+
+		if (player !is null)
+		{
+			bool myPlayer = player.isMyPlayer();
+			
+			switch (selected)
+			{
+				case 0: break;
+				case 1: mvm.VotedCount1++; break;
+				case 2: mvm.VotedCount2++; break;
+				case 3: mvm.VotedCount3++; break;
+			}
+			switch (lastselected)
+			{
+				case 0: break;
+				case 1: mvm.VotedCount1--; break;
+				case 2: mvm.VotedCount2--; break;
+				case 3: mvm.VotedCount3--; break;
+			}
+
+			if (myPlayer) current_Selected = selected;
+		}
+	}
+	else if (getNet().isServer() && cmd == this.getCommandID(vote_end_id))
+	{		
+		SaveVoteStats(this);
```

Comment:
Instead of reading/writing the stats to a cfg. Just dump the vote counts to rcon.
MapVotes map1name map1votes map2name map2votes random/novotes